raster/r.mfilter/testsuite/test_r_mfilter.py

```diff
@@ -184,10 +184,10 @@ class TestNeighbors(TestCase):
 
     def create_filter(self, options):
         """Create a temporary filter file with the given name and options."""
-        f = NamedTemporaryFile()
-        f.write(options)
-        f.flush()
-        return f
+        grass_tempfile = gs.tempfile(create=False)
+        with open(grass_tempfile, "wb") as f:
+            f.write(options)
+        return grass_tempfile
```

Comment:
I still don't get what's wrong with NamedTemporaryFile.

Reply:
I am a bit puzzled by that as well. `NamedTemporaryFile` leads to errors in the OSGeo4W tests. See:
https://github.com/OSGeo/grass/actions/runs/3368434482/jobs/5586972317#step:9:3031

I guess there is some _cygwin_-path-wrangling going on, maybe. At least the `C:\\Users\\RUNNER~1` short form in the path string is not working. If the GRASS GIS DB is in the users home directory, the grass.tempfile may fail as well with the same issue... It is jus a try (as I am not coding on Windows)... If that fails too, we may revert and try to resolve the Path in a way that it is not shortened....

Reply:
Putting together the [NamedTemporaryFile](https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile) doc and [this SE](https://stackoverflow.com/questions/18903069/how-can-i-read-namedtemporaryfile-in-python), I think that we need to use delete=False, write to the file, close it, and then delete it manually.